### PR TITLE
Add support for additional secret, initContainers, postStart command and deployment annotations

### PR DESCRIPTION
**Pull request by faizanahmad055**
Signed-off-by: Faizan Ahmad
username: faizanahmad055
email: faizan.ahmad55@outlook.com

#### What this PR does / why we need it:
This PR adds following in sonatype-nexus chart
- support for additional secrets
- support for init containers in the deployment
- support for postStart command in nexus container
- support for deployment annotations
- support for timeoutSeconds for livenessProb and readinessProbe

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by faizanahmad055** [comment]
Hi @rjkernick, Can you please take a look.

**Comment by faizanahmad055** [comment]
Hi @rjkernick ,

I've pushed one more commit can you please take a look.
This commit includes
- Support for additional containers
- Support for additional volumes
- Separate Exposed port and nexus-proxy container port.

The purpose of adding additional container and volumes is to provide the support of running extra sidecars. For example, I want to run keycloak container for nexus SSO authentication so I can add this using additional containers and volumes.

File: stable/sonatype-nexus/README.md [hunk 1]
```diff
@@ -77,13 +77,16 @@ The following table lists the configurable parameters of the Nexus chart and the
...
 | `nexus.readinessProbe.initialDelaySeconds`  | ReadinessProbe initial delay        | 30                                      |
 | `nexus.readinessProbe.periodSeconds`        | Seconds between polls               | 30                                      |
 | `nexus.readinessProbe.failureThreshold`     | Number of attempts before failure   | 6                                       |
+| `nexus.readinessProbe.timeoutSeconds`       | Time in seconds after readiness probe times out    | `nil`                    |
 | `nexus.readinessProbe.path`                 | Path for ReadinessProbe             | /                                       |
 | `nexus.hostAliases`                         | Aliases for IPs in /etc/hosts       | []                                      |
-| `nexusProxy.port`                           | Port for exposing Nexus             | `8080`                                  |
+| `nexusProxy.port`                           | Container Port for Nexus proxy      | `8080`                                  |
+| `nexusProxy.exposedPort`                    | Port for exposing Nexus             | `8080`                                  |
```
Comment (rjkernick) [inline stable/sonatype-nexus/README.md]: I'd rather see this as port and targetPort to match the yaml

**Review by rjkernick — CHANGES_REQUESTED** [review]

**Review by rjkernick — APPROVED** [review]

**Pull request merged**